core/src/main/java/org/geysermc/geyser/entity/type/ThrowableItemEntity.java

```diff
@@ -48,29 +49,38 @@ public ThrowableItemEntity(GeyserSession session, int entityId, long geyserId, U
         super(session, entityId, geyserId, uuid, definition, position, motion, yaw, pitch, headYaw);
         setFlag(EntityFlag.INVISIBLE, true);
         invisible = false;
+        age = 0;
+    }
+
+    @Override
+    protected void initializeMetadata() {
+        super.initializeMetadata();
+        // Correct sizing
+        dirtyMetadata.put(EntityDataTypes.SCALE, 0.5f);
     }
 
     private void checkVisibility() {
+        age++;
+
+        // Prevent projectiles from blocking the player's screen
+        if (session.isTickingFrozen()) {
+            // This may seem odd, but it matches java edition
+            Vector3f playerPos = session.getPlayerEntity().getPosition().sub(0, session.getPlayerEntity().getDefinition().offset(), 0);
```

Comment:
you can use `.down()` instead, looks a bit cleaner 
```suggestion
            Vector3f playerPos = session.getPlayerEntity().getPosition().down(EntityDefinitions.PLAYER.offset());
```